matchzoo/models/arci_model.py

```diff
@@ -0,0 +1,100 @@
+"""An implementation of ArcI Model."""
+from matchzoo import engine
+import numpy as np
+from keras.models import Model
+from keras.layers import Input, Embedding, Conv1D, MaxPooling1D, Dropout, \
+                         Concatenate, Flatten
+
+
+class ArcIModel(engine.BaseModel):
+    """
+    ArcI Model.
+
+    Examples:
+        >>> model = ArcIModel()
+        >>> model.guess_and_fill_missing_params()
+        >>> model.build()
+
+    """
+
+    @classmethod
+    def get_default_params(cls) -> engine.ParamTable:
+        """:return: model default parameters."""
+        params = super().get_default_params()
+        params['optimizer'] = 'adam'
+        params['input_shapes'] = [(32,), (32,)]
+        params['input_dtypes'] = [np.int32, np.int32]
+        params.add(engine.Param('trainable_embedding', False))
+        params.add(engine.Param('dim_embedding', 300))
+        params.add(engine.Param('vocab_size', 100))
+        params.add(engine.Param('num_blocks', 1))
+        params.add(engine.Param('left_kernel_count', [32]))
+        params.add(engine.Param('left_kernel_size', [3]))
+        params.add(engine.Param('right_kernel_count', [32]))
+        params.add(engine.Param('right_kernel_size', [3]))
+        params.add(engine.Param('activation', 'relu'))
+        params.add(engine.Param('left_pool_size', [16]))
+        params.add(engine.Param('right_pool_size', [16]))
+        params.add(engine.Param('padding', 'same'))
+        params.add(engine.Param('dropout_rate', 0.0))
+        params.add(engine.Param('embedding_mat',
+                   np.random.uniform(-0.2, 0.2, (params['vocab_size'],
+                                                 params['dim_embedding']))))
+        assert(params['num_blocks'] == len(params['left_kernel_count']))
+        assert(params['num_blocks'] == len(params['left_kernel_size']))
+        assert(params['num_blocks'] == len(params['left_pool_size']))
+        assert(params['num_blocks'] == len(params['right_kernel_count']))
+        assert(params['num_blocks'] == len(params['right_kernel_size']))
+        assert(params['num_blocks'] == len(params['right_pool_size']))
+        return params
+
+    def _conv_pool_block(self, input, kernel_count: int, kernel_size: int,
+                         padding: str, activation: str, pool_size: int):
```

Comment:
isn't `kernel_count`, `kernel_size`, `padding`, `activation`, `pool_size`  can be get from `self._params`?

Reply:
left and right have different parameters.